internal/providers/terraform/dir_provider.go

```diff
@@ -354,7 +354,7 @@ func (p *DirProvider) runInit(opts *CmdOptions, spinner *ui.Spinner) error {
 	if p.IsTerragrunt {
 		args = append(args, "run-all", "--terragrunt-ignore-external-dependencies")
 	}
-	args = append(args, "init", "-input=false", "-no-color")
+	args = append(args, "init", "-input=false", "-no-color", "-upgrade")
```

Comment:
Had to add this flag, otherwise the locked v4 is used for v3 tests.

Reply:
We might want to do this conditionally just for the tests. I don't think we should be upgrading the user's modules by default.

Reply:
Yep, you're right. I'll add it when `IsTest()` is true only. 👍

Reply:
Fixed in https://github.com/infracost/infracost/pull/1370/commits/9f7fc048cf65ad74a6c458e4d1d355f387c5031f.